src/main/java/de/plushnikov/intellij/plugin/processor/ValProcessor.java

```diff
@@ -104,6 +90,17 @@ private static boolean isSameName(String className) {
     return LOMBOK_VAL_SHORT_NAME.equals(className) || LOMBOK_VAL_FQN.equals(className);
   }
 
+  public Boolean hasModifierProperty(@NotNull PsiModifierList modifierList, @NotNull String name) {
+    final PsiElement parent = modifierList.getParent();
+
+    //Only apply final modifier to val local variables
+    if (parent instanceof PsiLocalVariable && isVal((PsiLocalVariable) parent)) {
+      return PsiModifier.FINAL.equals(name) ? Boolean.TRUE : null;
```

Comment:
It could be better to check for PsiModifier.FINAL.equals(name) first, before checking for instanceof and isVal